src/shared/awsContext.ts

```diff
@@ -54,6 +58,38 @@ export class DefaultAwsContext implements AwsContext {
 
         const persistedRegions = context.globalState.get<string[]>(regionSettingKey)
         this.explorerRegions = persistedRegions || []
+
+        this.lastTouchedRegion = 'None'
+    }
+
+    public guessDefaultRegion(node?: AWSTreeNodeBase): string {
+        /*
+        Checks in order of precedence
+       1. the node passed to the current command
+       2. last-expanded AWS Explorer service / etc
+       3. last selection region in RegionProvider of Wizard. 
+       3. final fallback is the credentials default region
+        */
+        if (this.explorerRegions.length === 1) {
+            return this.explorerRegions[0]
+        } else if (node) {
+            return node.determineRegion()
```

Comment:
should be able to do something like...

```suggestion
        } else if (node && (node as any).regionCode) {
            return node.regionCode
```

then we don't need all of the `determineRegion()` changes.

Reply:
I originally was just going to check if it had a regionCode parameter and if it didn't, then give up since not all of the nodes store this information in regionCode, some call it region, and some don't have it. However, I realized you can do better since there are many Node classes that do not have a regionCode property, but can have a determineRegion method, by either referring to its parent Node or children, or in rare cases another property. Having the method gives every Node a region that can be determined.